snippets/firestore/firestore_pipelines.py

```diff
@@ -0,0 +1,1289 @@
+    for res in result:
+        print(res)
+
+def timestamp_sub_function():
+    # [START timestamp_sub]
+    result = client.pipeline() \
+        .collection("documents") \
+        .select(
+            Field.of("expiresAt").timestamp_subtract("day", 14).as_("sendWarningTimestamp")
+        ) \
+        .execute()
+    # [END timestamp_sub]
+    for res in result:
+        print(res)
+
+def timestamp_to_unix_micros_function():
+    # [START timestamp_unix_micros]
+    result = client.pipeline() \
+        .collection("documents") \
+        .select(
+            Field.of("dateString").timestamp_to_unix_micros().as_("unixMicros")
+        ) \
+        .execute()
+    # [END timestamp_unix_micros]
+    for res in result:
+        print(res)
+
+def timestamp_to_unix_millis_function():
+    # [START timestamp_unix_millis]
+    result = client.pipeline() \
+        .collection("documents") \
+        .select(
+            Field.of("dateString").timestamp_to_unix_millis().as_("unixMillis")
+        ) \
+        .execute()
+    # [END timestamp_unix_millis]
+    for res in result:
+        print(res)
+
+def timestamp_to_unix_seconds_function():
+    # [START timestamp_unix_seconds]
+    result = client.pipeline() \
+        .collection("documents") \
+        .select(
+            Field.of("dateString").timestamp_to_unix_seconds().as_("unixSeconds")
+        ) \
+        .execute()
+    # [END timestamp_unix_seconds]
+    for res in result:
+        print(res)
+
+def cosine_distance_function():
+    # [START cosine_distance]
+    sample_vector = [0.0, 1.0, 2.0, 3.0, 4.0, 5.0]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(
+            Field.of("embedding").cosine_distance(sample_vector).as_("cosineDistance")
+        ) \
+        .execute()
+    # [END cosine_distance]
+    for res in result:
+        print(res)
+
+def dot_product_function():
+    # [START dot_product]
+    sample_vector = [0.0, 1.0, 2.0, 3.0, 4.0, 5.0]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(
+            Field.of("embedding").dot_product(sample_vector).as_("dotProduct")
+        ) \
+        .execute()
+    # [END dot_product]
+    for res in result:
+        print(res)
+
+def euclidean_distance_function():
+    # [START euclidean_distance]
+    sample_vector = [0.0, 1.0, 2.0, 3.0, 4.0, 5.0]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(
+            Field.of("embedding").euclidean_distance(sample_vector).as_("euclideanDistance")
+        ) \
+        .execute()
+    # [END euclidean_distance]
+    for res in result:
+        print(res)
+
+def vector_length_function():
+    # [START vector_length]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(
+            Field.of("embedding").vector_length().as_("vectorLength")
+        ) \
+        .execute()
+    # [END vector_length]
+    for res in result:
+        print(res)
```

Comment:
We usually use the [`black` code formatter](https://pypi.org/project/black/) for our Python code. If you haven't used it, it might be good to run it over this file to see if it has any changes

Nothing really stands out to me as badly formatted, and I don't think using that formatting should be a requirement for samples. But it might be worth checking